dhis-2/dhis-services/dhis-service-analytics/src/test/java/org/hisp/dhis/analytics/event/data/EventAnalyticsServiceTest.java

```diff
@@ -145,100 +132,202 @@ public void setUpTest()
-
-        org.hisp.dhis.trackedentity.TrackedEntityInstance maleA = createTrackedEntityInstance( ouA );
-        maleA.setUid( "person1234A" );
-        org.hisp.dhis.trackedentity.TrackedEntityInstance femaleB = createTrackedEntityInstance( ouB );
-        femaleB.setUid( "person1234B" );
-
-        maleA.setTrackedEntityType( trackedEntityType );
-        femaleB.setTrackedEntityType( trackedEntityType );
-
-        idObjectManager.save( maleA );
-        idObjectManager.save( femaleB );
+    @Override
+    public void tearDownTest()
+    {
+        analyticsTableGenerator.dropTables();
+    }
 
-        programInstanceService.enrollTrackedEntityInstance( maleA, programA, null, null, ouA );
-        programInstanceService.enrollTrackedEntityInstance( femaleB, programA, null, null, ouA );
+    /**
+     * <p>
+     * To create a new test that needs the analytics table populated:
+     * <p>
+     * <ul>
+     * <li>Make new EventQueryParam.</li>
+     * <li>Add to 'eventQueryParams' map.</li>
+     * <li>Add HashMap<String, Double> with expected output to results map.</li>
+     * </ul>
+     */
+    @Test
+    public void testGridAggregation()
+        throws IOException
+    {
+        // Given
 
-        // Read event data from CSV file
+        // Stubbed events data
+        stubAnalyticsEventsData();
 
+        // Events data from CSV file
         List<String[]> eventDataLines = CsvUtils.readCsvAsListFromClasspath( "csv/eventData.csv", true );
         parseEventData( eventDataLines );
 
-        // Generate analytics tables
-
+        // The generated analytics tables
         analyticsTableGenerator.generateTables( AnalyticsTableUpdateParams.newBuilder().build() );
 
-        // Set parameters
+        // The user
+        createAndInjectAdminUser();
 
-        // all events in program A - 2017
+        // All events in program A - 2017
         EventQueryParams events_2017_params = new EventQueryParams.Builder()
             .withOrganisationUnits( Lists.newArrayList( ouA ) )
             .withStartDate( getDate( 2017, 1, 1 ) )
             .withEndDate( getDate( 2017, 12, 31 ) )
             .withProgram( programA )
             .build();
 
+        // The params
+        Map<String, EventQueryParams> eventQueryParams = new HashMap<>();
         eventQueryParams.put( "events_2017", events_2017_params );
 
-        // Set results
-        // --------------------------------------------------------------------
-
+        // The results
         Map<String, Double> events_2017_keyValue = new HashMap<>();
         events_2017_keyValue.put( "ouabcdefghA", 6.0 );
 
+        Map<String, Map<String, Double>> results = new HashMap<>();
         results.put( "events_2017", events_2017_keyValue );
 
-        createAndInjectAdminUser();
-    }
+        for ( Map.Entry<String, EventQueryParams> entry : eventQueryParams.entrySet() )
+        {
+            String key = entry.getKey();
+            EventQueryParams params = entry.getValue();
```

Comment:
The for loop was in the code only because it was patterned after `AnalyticsServiceTest` which combined multiple test cases into a single JUnit test. If you only have a single test case, I believe you could replace:

        // The params
        Map<String, EventQueryParams> eventQueryParams = new HashMap<>();
        eventQueryParams.put( "events_2017", events_2017_params );

        // The results
        Map<String, Double> events_2017_keyValue = new HashMap<>();
        events_2017_keyValue.put( "ouabcdefghA", 6.0 );

        Map<String, Map<String, Double>> results = new HashMap<>();
        results.put( "events_2017", events_2017_keyValue );

        for ( Map.Entry<String, EventQueryParams> entry : eventQueryParams.entrySet() )
        {
            String key = entry.getKey();
            EventQueryParams params = entry.getValue();

            // When
            Grid aggregatedDataValueGrid = eventAnalyticsService.getAggregatedEventData( params );

            // Then
            AnalyticsTestUtils.assertResultGrid( aggregatedDataValueGrid, results.get( key ) );
        }

with something like:

        // The results
        Map<String, Double> events_2017_keyValue = new HashMap<>();
        events_2017_keyValue.put( "ouabcdefghA", 6.0 );

        // When
        Grid aggregatedDataValueGrid = eventAnalyticsService.getAggregatedEventData( events_2017_params );

        // Then
        AnalyticsTestUtils.assertResultGrid( aggregatedDataValueGrid, events_2017_keyValue );

However I still think we may want to have multiple test cases inside a single JUnit test. (See my next comment.)